include/integratorxx/atomic_densities/SlaterAtomicShell.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+#include <vector>
+#include <cmath>
+
+class SlaterTypeAtomicShell {
+  /// Angular momentum
+  int angular_momentum;
+  /// Exponents
+  std::vector<double> exponents;
+  /// Principal quantum numbers
+  std::vector<int> quantum_numbers;
+  /// Contraction orbital_coefficients
+  std::vector<std::vector<double>> orbital_coefficients;
+  /// Alpha orbital occupations
+  std::vector<int> alpha_occupations;
+  /// Beta orbital occupations
+  std::vector<int> beta_occupations;
+ public:
+  /// Dummy constructor
+  SlaterTypeAtomicShell() {};
+  /// Constructor
+  SlaterTypeAtomicShell(int angular_momentum_, const std::vector<double> & exponents_, const std::vector<int> & quantum_numbers_, const std::vector<std::vector<double>> & coefficients_, const std::vector<int> & alpha_occupations_, const std::vector<int> & beta_occupations_) : angular_momentum(angular_momentum_), exponents(exponents_), quantum_numbers(quantum_numbers_), orbital_coefficients(coefficients_), alpha_occupations(alpha_occupations_), beta_occupations(beta_occupations_) {};
+  /// Deconstructor
+  ~SlaterTypeAtomicShell() {};
+
+  /// Evaluates the basis functions
+  std::vector<double> evaluate_basis_functions(double r) {
```

Comment:
Returning `std::vector` is an anti-pattern, we should always expose the capability to populate preallocated memory. I'd probably opt to have something like [`gau2grid`](https://github.com/dgasmith/gau2grid) for Slaters, but of course with a lot less logic / having the ability to trigger fast code paths for spherically averaged functions

Reply:
OK, I've exposed the capabilities at least partly. Have a look